test/scenarios/fork/ynETHWithdrawals.t.sol

```diff
@@ -0,0 +1,544 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import {IWithdrawalQueueManager} from "../../../src/interfaces/IWithdrawalQueueManager.sol";
+import {IRedeemableAsset} from "../../../src/interfaces/IRedeemableAsset.sol";
+import {IRedemptionAssetsVault} from "../../../src/interfaces/IRedemptionAssetsVault.sol";
+import {IynETH} from "../../../src/interfaces/IynETH.sol";
+
+import {StakingNode} from "../../../src/StakingNode.sol";
+import {WithdrawalQueueManager} from "../../../src/WithdrawalQueueManager.sol";
+import {ynETHRedemptionAssetsVault, ETH_ASSET} from "../../../src/ynETHRedemptionAssetsVault.sol";
+
+import "../../utils/StakingNodeTestBase.sol";
+
+contract ynETHWithdrawals is StakingNodeTestBase {
+
+    error CallerNotOwnerNorApproved(uint256 tokenId, address caller);
+    error NotFinalized(uint256 currentTimestamp, uint256 requestTimestamp, uint256 queueDuration);
+    error ERC721NonexistentToken(uint256 tokenId);
+    error InsufficientBalance(uint256 currentBalance, uint256 requestedBalance);
+    error InsufficientAssetBalance(address asset, uint256 requestedAmount, uint256 balance);
+    error AccessControlUnauthorizedAccount(address account, bytes32 neededRole);
+    error AmountExceedsSurplus(uint256 requestedAmount, uint256 availableSurplus);
+
+    bool public isHolesky;
+
+    address public constant user = address(0x12345678);
+    address public constant receiver = address(0x987654321);
+
+    uint256 public nodeId = 2;
+    uint256 public withdrawalAmount = 32 ether;
+
+    uint256 tokenId;
+    uint256 secondsToFinalization;
+    uint256 amountOut;
+
+    StakingNode public stakingNode;
+
+    // ------------------------------------------
+    // Setup
+    // ------------------------------------------
+
+    function setUp() public override {
+        super.setUp();
+
+        isHolesky = block.chainid == 17000;
+
+        if(isHolesky) {
+            stakingNode = StakingNode(payable(address(stakingNodesManager.nodes(nodeId))));
+
+            vm.deal(user, 10_000 ether);
+
+            setupForVerifyWithdrawalCredentials(nodeId, "test/data/holesky_wc_proof_1916455.json");
+
+            secondsToFinalization = 1 days;
+            vm.prank(actors.ops.WITHDRAWAL_MANAGER);
+            ynETHWithdrawalQueueManager.setSecondsToFinalization(secondsToFinalization);
+        }
+    }
+
+    // ------------------------------------------
+    // Withdrawal flow tests
+    // ------------------------------------------
+
+    function testVerifyWithdrawalCredentials() public {
+        if (!isHolesky) return;
+
+        ValidatorProofs memory _validatorProofs = getWithdrawalCredentialParams();
+
+        vm.expectRevert(bytes4(keccak256("NotStakingNodesOperator()")));
```

Comment:
do not mix error cases with success cases in the same test.

They are separate tests

Reply:
fixed in https://github.com/yieldnest/yieldnest-protocol/pull/124/commits/e40cb6f14cb1feb306eafa7a9ba859b03944625e.